### DISCUSSION TITLE
V5 - Feature Request - make `_systemId` Public

### DISCUSSION THREAD

**Discussion by christophe-g** [statelyai/xstate#4544] [Ideas] [answered]
First of all, thanks a lot for the recent release of version 5 - The code is great and the type inference system is quite something!

Now to this feature request: is it possible to make [`actor._systemId`](https://github.com/statelyai/xstate/blob/eef14a2d558a00ad75558fb44432601608690890/packages/core/src/interpreter.ts#L120) public? 

My use case for requesting this is the following: 
- There is a machine running on the cloud, and persisted on db. 
- When this machine runs or invoke other actors, I'd like to pass on the `_systemId` of the root machine, which is the id of the persisted actor (in this case). For the time being, it is not possible without tricking TS.

For instance, a machine handling a user invite to a service will send an email to this user. I would like the email to have some meta data re which machine instance this was originated from, for instance, when the sendEmail actor is invoked: 

```js
 invoke: {
    scr: 'sendEmail',
    input: ({context: {message, to}, {self: {_systemId}}}) => ({message, to, {actorID: _systemId}})
}
```

This could be part of the `context` you might say, but imo it does not belong here as it does not affect the way the machine runs or work. 

I can send a PR if this makes sense. 

Otherwise, just close this and I will go with the `context` route

**Comment by Andarist** [comment, marked as answer]
Could you prepare a more full example of this scenario? I'd like to understand better how you instantiate this system etc.
